### ISSUE TITLE
vmagent relabeling error re-writing regex

### ISSUE THREAD

**Issue by cskinfill** [VictoriaMetrics/VictoriaMetrics#3131]
**Describe the bug**
Regression when vmagent loads a relabeling config it is failing to parse regex because it is incorrectly rewriting it.

**To Reproduce**
It appears to be any relabeling regex with an escaped `$` (so `\$`) at the end of the regex string.

For example

```
- source_labels: [__name__]
  action: replace
  regex: '(.*)\$$'
  target_label: "__name__"
  replacement: "$1"
```

**Expected behavior**
The relabeling config will continue to work.

**Logs**
```
2022-09-19T17:37:16.055Z	fatal	VictoriaMetrics/app/vmagent/remotewrite/remotewrite.go:139	cannot load relabel configs: cannot load -remoteWrite.relabelConfig="/etc/vm/relabeling/global_relabeling.yaml": cannot unmarshal `relabel_configs` from "/etc/vm/relabeling/global_relabeling.yaml": error when parsing `relabel_config` #5: cannot parse `regex` "^(?:(.*)\\)$": error parsing regexp: missing closing ): `^(?:(.*)\)$`
```

**Version**
The line returned when passing `--version` command line flag to the binary. For example:
```
vmagent-20220908-110742-tags-v1.81.2-0-gdd1126dcc
```

**Used command-line flags**
```
-graphiteListenAddr=":2003"
-httpListenAddr=":8429"
-influx.maxLineSize="8MB"
-promscrape.cluster.memberNum="1"
-promscrape.cluster.membersCount="5"
-promscrape.config="/etc/vmagent/config_out/vmagent.env.yaml"
-promscrape.maxScrapeSize="1GiB"
-promscrape.streamParse="true"
-promscrape.suppressDuplicateScrapeTargetErrors="true"
-promscrape.suppressScrapeErrors="true"
-remoteWrite.maxDiskUsagePerURL="1073741824"
-remoteWrite.relabelConfig="/etc/vm/relabeling/global_relabeling.yaml"
-remoteWrite.tmpDataPath="/tmp/vmagent-remotewrite-data"
-remoteWrite.url="secret"
```

It appears to be this function https://github.com/VictoriaMetrics/VictoriaMetrics/blob/e9a0b803a4c5bcf31587f26cd0b5bdb84cd032cb/lib/regexutil/regexutil.go#L11 which will loop and remove any `$` in the regex string or substrings.

**Comment by valyala** [comment]
The issue should be fixed in the commit fda60b3d4d22e76ced2f3ba761c72552d3a5c6a9 . This commit will be included in the next release. In the mean time it is possible to build vmagent from this commit according to [these instructions](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources) and verify whether it correctly handles `regex` ending with escaped `$` in relabeling rules.

**Comment by valyala** [comment]
`vmagent` should properly handle escaped `$` char at the end of `regex` in relabeling configs starting from [v1.82.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
